### PR TITLE
fixed Injury Log Text; still incomplete

### PR DESCRIPTION
**Pull request by SzieberthAdam**
Most of the Armor/Injury/Casualty reports are done. Final defender state and injury reports are missing only.

For those, `client.StatusReport.reportInjury()` yet to be converted. It requires a converted `PlayerState` class which is also missing yet.

4a8faf5 and d05c4dc are fixes. Unity weirdly not generated errors for a short time and I thought my conversion is fine but it was not.

### CODE REVIEW

**Comment by SzieberthAdam** [comment]
I spent about 2 hours to convert report.seriousInjury to Fumbbl.Model.Types.SeriousInjury without success. I could have make SeriousInjuries public but thought that it must be a better way.

File: Assets/Scripts/Model/RollModifier/AbstractModifier.cs [hunk 1]
```diff
@@ -5,25 +5,44 @@ namespace Fumbbl.Model.RollModifier
...
+                    }
+                    else
+                    {
+                        return $" {Name}";
+                    }
+
                 }
 
-                string sign = Modifier <= 0 ? "+" : "-";
+                string sign = string.Empty;
```
Comment (christerk) [inline Assets/Scripts/Model/RollModifier/AbstractModifier.cs:29]: First, as a style and performance guideline, you shouldn't be using string addition in a situation like this. leave the intializer off and use sign = below.

That being said, I would be happier to replace this processing with the more elegant:

if (ReversedModifier)
{
  Modifier = -Modifier;
}

File: Assets/Scripts/Model/RollModifier/AbstractModifier.cs [hunk 2]
```diff
@@ -5,25 +5,44 @@ namespace Fumbbl.Model.RollModifier
     public abstract class AbstractModifier : FfbEnumerationFactory
     {
         public int Modifier { get; set; }
+        public bool ForcedModifiedString { get; set; } = false;
```
Comment (christerk) [inline Assets/Scripts/Model/RollModifier/AbstractModifier.cs]: This is not an elegant solution. The functional equivalent would be to have a "SignIncludedInName" property together with the two other configuration properties.

It's probably a good idea to add a straight-up format datatype to group these settings:
struct Format {
  bool ShowSign;
  bool ShowModifier;
  bool ShowName;
  bool ReverseModifier;
}

Makes the processing clearer and more generic in nature.

File: Assets/Scripts/Model/Types/PlayerState.cs [hunk 3]
```diff
@@ -45,5 +45,30 @@ public static PlayerState Get(int? state)
 
             return PlayerStates[key];
         }
+
+        public string getDescription()
```
Comment (christerk) [inline Assets/Scripts/Model/Types/PlayerState.cs]: Change this to a property instead:

public string Description => {
  switch ...
}

**Review by christerk — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Model/Types/PlayerState.cs (hunk 3 above)
Comment (SzieberthAdam) [reply Assets/Scripts/Model/Types/PlayerState.cs]: Was not able to do it with lambda without errors. Ended up with a regular getter instead.

**Review by SzieberthAdam — COMMENTED** [review]
Done with the requested changes. Another hour spent on SeriousInjury conversion...

PlayerState.cs: Was not able to do it with lambda without errors. Ended up with a regular getter instead.

**Comment by SzieberthAdam** [comment]
Done with SeriousInjury Type and Injury report.

Some remarks:

- I mimicked KickOffResult the best I could.
- ApothecaryChoice,cs was also affected. As a note I am suspicious that its DTO's seriousInjury value is always null.
- `using Fumbbl.Model.Types;` in Injury.cs and ApothecaryChoice.cs resulted errors which weirdly disappeared after I removed the using.
- I had to handle null inputs in T As<T>.

File: Assets/Scripts/Model/FfbEnumerationFactory.cs [hunk 4]
```diff
@@ -43,6 +43,9 @@ public static void RegisterModifier(Type type, string key, FfbEnumerationFactory
         public static T As<T>(this FFBEnumeration ffbEnum)
             where T : FfbEnumerationFactory
         {
+            if ((ffbEnum is null) || (ffbEnum.key is null)){
```
Comment (christerk) [inline Assets/Scripts/Model/FfbEnumerationFactory.cs]: A quick style comment - The following would look cleaner for this line:
 if (ffbEnum?.key is null) {

File: Assets/Scripts/Model/RollModifier/InjuryModifier.cs [hunk 5]
```diff
@@ -4,16 +4,16 @@ public class InjuryModifier : AbstractModifier
     {
         public InjuryModifier(string name, int modifier) : base(name, modifier) { }
 
-        public bool NigglingInjuryModifier { get; set; }
+        public override ShowModifierSignMode ShowModifierSign { get; set; } = ShowModifierSignMode.ShowReversed;
```
Comment (christerk) [inline Assets/Scripts/Model/RollModifier/InjuryModifier.cs]: Seems excessive to override this property to change the default value. Set the default in the constructor instead.

**Comment by SzieberthAdam** [comment]
Note to 
2e2ef6d: https://gamedev.stackexchange.com/questions/50187/are-sln-csproj-usefull-for-sharing-the-unity-project

I needed to delete them to get my IDE able to identify the UnicodeEditor module. It gets regenerated anyway by Unity Editor.

**Comment by SzieberthAdam** [comment]
This happened with a TTM. Needs testing.

```
  at (wrapper castclass) System.Object.__castclass_with_cache(object,intptr,intptr)
  at System.Linq.Enumerable+<CastIterator>d__34`1[TResult].MoveNext () [0x00040] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at System.Linq.Enumerable.Contains[TSource] (System.Collections.Generic.IEnumerable`1[T] source, TSource value, System.Collections.Generic.IEqualityComparer`1[T] comparer) [0x00036] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at System.Linq.Enumerable.Contains[TSource] (System.Collections.Generic.IEnumerable`1[T] source, TSource value) [0x0000a] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at Fumbbl.UI.LogText.ThrowTeamMateRoll+<Convert>d__0.MoveNext () [0x00168] in C:\Users\Szieberth\Documents\Repos\FFBUnity\Assets\Scripts\UI\LogText\ThrowTeamMateRoll.cs:25 
  at TextPanelHandler.AddReport (Fumbbl.Ffb.Dto.Report report) [0x0005c] in C:\Users\Szieberth\Documents\Repos\FFBUnity\Assets\TextPanelHandler.cs:89 
  at (wrapper delegate-invoke) <Module>.invoke_void_Report(Fumbbl.Ffb.Dto.Report)
  at Fumbbl.FFB.TriggerLogChanged (Fumbbl.Ffb.Dto.Report text) [0x00002] in C:\Users\Szieberth\Documents\Repos\FFBUnity\Assets\Scripts\FFB.cs:123
```

**Pull request merged**